lib/libzutil/os/linux/zutil_device_path_os.c

```diff
@@ -271,9 +271,12 @@ zfs_get_pci_slots_sys_path(const char *dev_name)
 		if (strcmp(address1, address2) == 0) {
 			/* Addresses match, we're all done */
 			free(address2);
+			if (path != NULL) {
+				free(path);
+				path = NULL;
+			}
```

Comment:
It seems to me removing the `free(tmp)` should be sufficient.  I don't see how `path` could ever be non-NULL here.  If we successfully allocate it below we always `break` and `return (path)`.  If we fail to allocate `path` and `continue` around the loop again then there's nothing to free.  Am I overlooking something?

Reply:
You are right. I have been looking at so many issues involving code that I ended up defending against the loop continuing after the allocation because I mistook it for other loops where that happens. After the examining >75 coverity reports, it is all starting to look the same to me. :/